### ISSUE TITLE
Maybe publicFolder could do what Amit wants?

### ISSUE THREAD

**Issue by scripting** [scripting/publicfolder#6]
I just read a [review of publicFolder](https://ol.amitgawande.com/2021/11/18/124922.html?title=publicfolderIsNotExactlyWhatIThoughtItWas) by Amit. 

Interesting. This is what I think he's saying, phrased as a feature request.

> I would like an option to tell PublicFolder not to delete a file on S3 if I delete it from the local folder.

Did I get it right?

**Comment by am1t** [comment]
Not actually. Here's the situation. I publish posts to my Drummer blog from multiple systems, including Windows. I have PublicFolder on one machine only. If I am posting from another machine, I upload the image directly to S3. Given the way PublicFolder works, it deletes the just added image.

So, here's how I would like PublicFolder to behave.

> If it sees a new file on S3 which is not in the local folder, it should bring the file to local. It's ok if delete is not handled, for me personally. Possibly an option to not delete files from S3.

**Comment by am1t** [comment]
@scripting Did you get a chance to put some thought behind this? And I hope I was clear about the behaviour I expect.

**Comment by scripting** [comment]
@amit it would be helpful to have a description as concise as possible, of the new functionality that would fit your needs.

**Comment by scripting** [comment]
@am1t -- i missed your comment from five days ago. Sorry. I am juggling so many different things. 

Now I have read it. ;-)

Here's the problem with that approach -- there would be no way to delete a file from within publicFolder. 

I don't think that's a deal-stopper, just wanted to point that out.

**Comment by am1t** [comment]
Yeah, you are right. I had a behaviour like Dropbox in mind where a file's lifecycle is tracked - deleted, modified etc. But that will unnecessarily complicate the functionality. I believe a flag to not delete a file on S3 can be a good option. For me, S3 is the primary source.

**Comment by scripting** [comment]
@am1t -- I implemented a new config option that allows you to tell PF not to delete files.

http://scripting.com/drummer/blog/2021/11/23/161903.html?title=newVersionOfPublicFolderApp

It's really important to keep the flow of suggestions coming. Never assume that I wouldn't do something to make a product work better for you. That's part of my process of improving software over time. Not just to serve my needs but to serve the needs of smart users like yourself.

**Comment by am1t** [comment]
Thank you, Dave, for implementing the change. I really appreciate all the efforts you put in. I can start using the app again! And I will report things that don't work, or I expect to work differently, as always :)

**Comment by am1t** [comment]
BTW, I am not certain if it was a miss and something you are keen on addressing, but the config parameter name `flDeleteEmabled ` looks misspelt? Shouldn't matter, but I wanted to put to your ears anyway, once.

**Comment by scripting** [comment]
it is misspelled. i noticed it after i completed the release. i want to fix it but i will keep supporting the misspelling, so you won't have to worry.
